entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -502,6 +503,51 @@ class Neo4jRepository(
         )
     }
 
+    fun getEntityTypes(): List<Map<String, Any>> {
+        // The match on geoProperties is specific since they are not fully supported
+        // (currently we only support location that is stored among the entity node attributes)
+        val query =
+            """
+                MATCH (entity:Entity)
+                OPTIONAL MATCH (entity)-[:HAS_VALUE]->(property:Property)
+                OPTIONAL MATCH (entity)-[:HAS_OBJECT]->(rel:Relationship)
+                RETURN labels(entity) as entityType, collect(distinct property.name) as propertyNames, 
+                    reduce(output = [], r IN collect(distinct labels(rel)) | output + r) as relationshipNames,
+                    count(entity.location) as entityWithLocationCount
+            """.trimIndent()
+
+        val result = session.query(query, emptyMap<String, Any>(), true).toList()
+        return result.map { rowResult ->
+            val entityWithLocationCount = (rowResult["entityWithLocationCount"] as Long).toInt()
+            val entityTypes = (rowResult["entityType"] as Array<String>)
+                .filter { !authorizationEntitiesTypes.plus("Entity").contains(it) }
+            entityTypes.map { entityType ->
+                mapOf(
+                    "entityType" to entityType,
+                    "properties" to (rowResult["propertyNames"] as Array<Any>).toSet(),
+                    "relationships" to (rowResult["relationshipNames"] as Array<Any>)
+                        .filter { it !in listOf("Attribute", "Relationship") }.toSet(),
+                    "geoProperties" to
+                        if (entityWithLocationCount > 0) setOf("https://uri.etsi.org/ngsi-ld/location") else emptySet()
```

Comment:
use a const from JsonLdUtils (will be useful when it changes, and it will in 1.3.1 or 1.4.1)